script.js

```diff
@@ -0,0 +1,236 @@
+        <div class="cart__item-summary">
+            <span class="cart__item-price">${item.price.toFixed(2)}</span>
+            <span class="cart__item-amount">x <span id="cart-${item.id}">${
+			item.quantity
+		}</span></span>
+        </div>
+    </div>
+    <button type="button" id="removeBtn" class="button button--remove">Usuń</button>`;
+		cartItem.innerHTML = cartItemContent;
+	}
+};
+
+// Function removes an item from the cart
+const removeButtonHandler = (e) => {
+	if (e.target.matches('#removeBtn')) {
+		const cartItem = e.target.closest('li');
+		$cartArr.forEach((el) => {
+			if (el.id == cartItem.id) {
+				if (el.quantity === 1) {
+					const index = $cartArr.findIndex((item) => item.id == cartItem.id);
+					$cartArr.splice(index, 1);
+					badgeHandler();
+					totalPriceHandler();
+					cartItem.remove();
+					if ($cartArr.length === 0) {
+						cartHandler();
+					}
+				} else {
+					// if there is more items then 1, function changes quantity of items by 1
+					el.quantity--;
+					const amount = document.querySelector(`#cart-${el.id}`);
+					amount.innerText = el.quantity;
+					badgeHandler();
+					totalPriceHandler();
+				}
+			}
+		});
+	}
+};
+
+// this function executes only if the first item is added to cart or the last item is removed from cart
+const cartHandler = () => {
+	if ($cartArr.length === 0) {
+		// when the last item from cart is removed, function removing total price information and render another paragraph
+		$cartEmpty = document.createElement('div');
+		$cartEmpty.classList.add('cart__empty');
+		$cartEmpty.setAttribute('id', 'empty');
+		$cartEmpty.innerHTML = '<p>Głodny?<br />Zamów naszą pizzę!</p>';
+		$cartOrder.appendChild($cartEmpty);
+		$cartTotal.remove();
+	} else if ($cartArr.length === 1) {
+		// when first item is added to the cart, removes a paragraph and adding info about total price of all items.
+		const cartEmptyToRemove = document.querySelector('#empty');
+		cartEmptyToRemove.remove();
+
+		$cartTotal = document.createElement('div');
+		$cartTotal.classList.add('cart__total');
+		$cartOrder.appendChild($cartTotal);
+		const cartTotalContent = `
+        <span>Do zapłaty:</span>
+        <span id="total-price"></span>`;
+		$cartTotal.innerHTML = cartTotalContent;
+	}
+};
+
+// function checking the amount of all items and assigns this value to the badge
+// the function is executed each time any button is pressed
+const badgeHandler = () => {
+	const badge = document.querySelector('#badge');
+	let badgeValue = 0;
+	for (let i = 0; i < $cartArr.length; i++) {
+		badgeValue = badgeValue + $cartArr[i].quantity;
+	}
+	badge.innerText = badgeValue;
+};
+
+// function checking total price of all items from the cart
+// the function is executed each time any button is pressed
+const totalPriceHandler = () => {
+	const totalPrice = document.querySelector('#total-price');
+	let totalPriceValue = 0;
+	for (let i = 0; i < $cartArr.length; i++) {
```

Comment:
raz wolisz `forEach` a raz `for`, warto sie trzymac jednego rozwiazania

Reply:
Będę pamiętał na przyszłość